salt/files/mysql-backup-to-s3.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+#
+# This file is managed by Dogsbody Technology Ltd.
+#   https://www.dogsbody.com/
+#
+# Description:  Backup MySQL databases and upload to AWS S3
+#
+# Usage:  $0
+#
+
+set -euo pipefail
+
+MYSQL="/usr/bin/mysql"
+MYSQLDUMP="/usr/bin/mysqldump"
+MYSQL_DEFAULTS_FILE="/home/sysadmin-tools/mysql-defaults.cnf"
+
+source /home/sysadmin-tools/aws-settings.local
+
+export AWS_ACCESS_KEY_ID
+export AWS_SECRET_ACCESS_KEY
+export AWS_DEFAULT_REGION
+
+if [ "${LOGNAME}" != "root" ]; then
+    echo "ERROR: Execution of ${0} stopped as not run by user root!"
+    exit 3
+fi
+
+if [ ! -x "${AWS_CLI}" ]; then
+    echo "Error: The aws executable is not installed"
+    exit 3
+fi
```

Comment:
I like this check because it double checks the environment matches what we are configuring in salt.
It also double checks the `aws` binary has been installed to the location we expect - this can change depending on how aws-cli was installed.